receiver.go

```diff
@@ -345,29 +330,25 @@ func (r *Receiver) setLastError(err error) {
 	r.lastErrorMu.Unlock()
 }
 
-func (r *Receiver) listenForMessage(ctx context.Context) (*amqp.Message, error) {
+func (r *Receiver) listenForMessage(ctx context.Context, handler amqpHandler) error {
 	ctx, span := r.startConsumerSpanFromContext(ctx, "sb.Receiver.listenForMessage")
 	defer span.End()
 
 	var receiver *amqp.Receiver
 	r.clientMu.RLock()
 	if r.receiver == nil {
-		return nil, r.connClosedError(ctx)
+		return r.connClosedError(ctx)
 	}
 	receiver = r.receiver
 	r.clientMu.RUnlock()
-	msg, err := receiver.Receive(ctx)
+	err := receiver.HandleMessage(ctx, func(message *amqp.Message) error {
+		return handler.Handle(ctx, message)
+	})
 	if err != nil {
 		tab.For(ctx).Debug(err.Error())
-		return nil, err
-	}
-
-	id := messageID(msg)
-	if idStr, ok := id.(string); ok {
-		span.AddAttributes(tab.StringAttribute("amqp.message.id", idStr))
```

Comment:
We don't need this now?

Reply:
Oops missed it.

Reply:
right, the whole internal listenForMessage is now folded into the middlware handler that does the translation from amqp to SB message type.